cmd/prometheus-postgresql-adapter/main.go

```diff
@@ -232,6 +233,21 @@ func initElector(cfg *config, db *sql.DB) *util.Elector {
 	return &scheduledElector.Elector
 }
 
+var writeRequestPool = sync.Pool{}
+
+func acquireWriteRequest() *prompb.WriteRequest {
+	res := writeRequestPool.Get()
+	if res == nil {
+		return new(prompb.WriteRequest)
+	}
+	return res.(*prompb.WriteRequest)
+}
+
+func releaseWriteRequest(b *prompb.WriteRequest) {
+	b.Reset()
```

Comment:
This call will actually create a new instance every time.
I'm a bit sceptical that using `sync.Pool` here reduces memory pressure. On the other side I'd be happy to see some benchmarks to prove that I'm wrong. @mikefaraponov  Can you produce some benchmark numbers with/without sync.Pool?

Reply:
I believe that instead of `b.Reset()` doing `b.Timeseries = b.Timeseries[:0]` would actually avoid recreating instance every time. However I think we still need to produce basic benchmarking numbers and do memory profiling to make sure we're on the right track.

Reply:
Sure, I'll show benchmarks as soon as possible.
Actually, func (m *WriteRequest) Reset()         { *m = WriteRequest{} }
Creating empty structures creates no allocation it's the same as to nil keys of this structure by hand

Reply:
I checked with some kind of model for this case 
Benchmark1/raw-4         	  701991	      1627 ns/op	     312 B/op	       7 allocs/op
Benchmark1/pool-4      	  721689	      1661 ns/op	     280 B/op	       6 allocs/op
Variant with pool a little bit faster and has less allocation, but not significantly, maybe it's premature optimization